Calculator/Operator.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Operator.swift
+//  Calculator
+//
+//  Created by Yetti on 2023/06/09.
+//
+
+enum Operator: Character, CaseIterable, CalculateItem {
+    case add = "+"
+    case subtract = "-"
+    case divide = "/"
+    case multiply = "*"
+    
+    func calculate(lhs: Double, rhs: Double) -> Double {
+        switch self {
+        case .add:
+            return add(lhs: lhs, rhs: rhs)
+        case .subtract:
+            return subtract(lhs: lhs, rhs: rhs)
+        case .divide:
+            return divide(lhs: lhs, rhs: rhs)
+        case .multiply:
+            return multiply(lhs: lhs, rhs: rhs)
+        }
+    }
+    
+    private func add(lhs: Double, rhs: Double) -> Double {
+        return lhs + rhs
+    }
+    
+    private func subtract(lhs: Double, rhs: Double) -> Double {
+        return lhs - rhs
+    }
+    
+    private func divide(lhs: Double, rhs: Double) -> Double {
+        guard rhs.isZero == false else { return Double.nan }
+        return lhs / rhs
```

Comment:
0으로 나누는 것을 방지하는 로직이군요! 👍 Double에는 다양한 프로퍼티들이 존재합니다. 수학적으로 0으로 나누면 무한대로 수렴하죠? 이를 이용해서 좀 더 재미있게 표현해 볼 수 있어요.
```suggestion
        let result = lhs / rhs
        return result.isInfinite ? .nan : result
```
`isInfinite`라는 프로퍼티를 통해 무한대인지 아닌지 체크하는 로직입니다. 이런 방법도 있으니 참고해보세요~ 😁